flake/dev/flake.nix

```diff
@@ -0,0 +1,51 @@
+{
+  description = ''
+    Private inputs for development purposes. These are used by the top level
+    flake in the `dev` partition, but do not appear in consumers' lock files.
+  '';
+
+  inputs = {
+    # This input is prefixed with 'dev-' to avoid shadowing the public flake's
+    # 'nixpkgs' input, allowing other dev inputs to follow it.
+    #
+    # NOTE: The only reason we specify a nixpkgs input at all here, is so the
+    #       other inputs can follow it.
+    # TODO: Replace
+    #           dev-nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    #       with
+    #           dev-nixpkgs = {
+    #             url = "path:../..";
+    #             inputs = {
+    #               "<PUBLIC_INPUT_1>".follows = "";
+    #               "<PUBLIC_INPUT_2>".follows = "";
+    #               "<PUBLIC_INPUT_3>".follows = "";
+    #             };
+    #           };
+    #       once 26.05 is released, giving non-flake end-users and Stylix
+    #       contributors two LTS releases to adopt Nix 2.26+ [1]. Note that
+    #       non-nixpkgs public flake inputs should be disabled.
```

Comment:
Note, the nix version requirements only really apply when bumping or otherwise modifying the dev lockfile.

In normal usage, flake-parts will evaluate the dev flake using flake-compat, not `builtins.getFlake`.

Therefore the real issue is flake-compat getting relative path support and flake-parts updating its flake-compat version.